delta-lake/delta-33x/src/main/scala/com/nvidia/spark/rapids/delta/delta33x/GpuDelta33xParquetFileFormat.scala

```diff
@@ -417,109 +581,21 @@ case class GpuDelta33xParquetFileFormat(
+      try {
+        if (rowIndexColumnOpt.isDefined) {
+          indexVectorTuples += (rowIndexColumnOpt.get.index -> rowIndexGpuCol.incRefCount())
         }
-      withResource(containsMarkedRows) { containsMarkedRows =>
-        GpuColumnVector.from(containsMarkedRows.castTo(DType.INT8), ByteType)
-      }
-    }
-  }
-
-  private object RapidsKeepMarkedRowsFilter extends RapidsRowIndexMarkingFiltersBuilder {
-
-    override def getFilterForEmptyDeletionVector(): RapidsRowIndexFilter = RapidsDropAllRowsFilter
-
-    override def getFilterForNonEmptyDeletionVector(
-      bitmap: RoaringBitmapArray): RapidsRowIndexFilter = new RapidsKeepMarkedRowsFilter(bitmap)
-  }
-
-  private object RapidsDropMarkedRowsFilter extends RapidsRowIndexMarkingFiltersBuilder {
-
-    override def getFilterForEmptyDeletionVector(): RapidsRowIndexFilter = RapidsKeepAllRowsFilter
-
-    override def getFilterForNonEmptyDeletionVector(
-      bitmap: RoaringBitmapArray): RapidsRowIndexFilter = new RapidsDropMarkedRowsFilter(bitmap)
-
-  }
-
-  private object RapidsDropAllRowsFilter extends RapidsRowIndexFilter {
-    def materializeIntoVector(rowIndexCol: GpuColumnVector): GpuColumnVector = {
-      withResource(Scalar.fromByte(1.toByte)) { one =>
-        GpuColumnVector.from(one, rowIndexCol.getRowCount.toInt, ByteType)
-      }
-    }
-  }
-
-  private object RapidsKeepAllRowsFilter extends RapidsRowIndexFilter {
-    def materializeIntoVector(rowIndexCol: GpuColumnVector): GpuColumnVector = {
-      withResource(Scalar.fromByte(0.toByte)) { zero =>
-        GpuColumnVector.from(zero, rowIndexCol.getRowCount.toInt, ByteType)
-      }
-    }
-  }
-
-  trait RapidsRowIndexMarkingFiltersBuilder {
-    def getFilterForEmptyDeletionVector(): RapidsRowIndexFilter
-    def getFilterForNonEmptyDeletionVector(bitmap: RoaringBitmapArray): RapidsRowIndexFilter
-
-    def createInstance(
-        deletionVector: DeletionVectorDescriptor,
-        hadoopConf: Configuration,
-        tablePath: Option[Path]): RapidsRowIndexFilter = {
-      if (deletionVector.cardinality == 0) {
-        getFilterForEmptyDeletionVector()
-      } else {
-        require(tablePath.nonEmpty, "Table path is required for non-empty deletion vectors")
-        val dvStore = new HadoopFileSystemDVStore(hadoopConf)
-        val storedBitmap = StoredBitmap.create(deletionVector, tablePath.get)
-        val bitmap = storedBitmap.load(dvStore)
-        getFilterForNonEmptyDeletionVector(bitmap)
+        startTime = System.nanoTime()
+        val isRowDeletedVector = rowIndexFilterOpt.get.materializeIntoVector(rowIndexGpuCol)
+        metrics("isRowDeletedColumnGenTime") += System.nanoTime() - startTime
+        indexVectorTuples += (isRowDeletedColumnOpt.get.index -> isRowDeletedVector)
+        replaceVectors(batch, indexVectorTuples.toSeq: _*)
+      } catch {
+        case e: Exception => indexVectorTuples.foreach(item => item._2.close())
```

Comment:
`close()` can throw an exception, and we'd want it to be reported as well.

```suggestion
        case e: Exception => 
          try {
            indexVectorTuples.foreach(item => item._2.close())
          } catch {
            case t: Throwable =>
              e.addSuppressed(t)
          }
```